graphql/resolvers/exerciseCrud.ts

```diff
@@ -91,3 +93,75 @@ export const deleteExercise = async (
     }
   })
 }
+
+export const flagExercise = async (
+  _parent: void,
+  arg: MutationFlagExerciseArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const { id, flagReason } = arg
+
+  const flaggedById = req.user?.id
+  if (!flaggedById) throw new Error('No User')
+
+  const exercise = await prisma.exercise.findUnique({
+    where: {
+      id
+    }
+  })
+
+  if (get(exercise, 'flaggedAt')) {
+    throw new Error('Exercise is already flagged')
+  }
+
+  return prisma.exercise.update({
+    where: { id },
+    data: {
+      flaggedAt: new Date().toISOString(),
+      flaggedById,
+      flagReason
+    },
+    include: {
+      author: true,
+      module: true
+    }
+  })
+}
+
+export const removeExerciseFlag = async (
+  _parent: void,
+  arg: MutationRemoveExerciseFlagArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const { id } = arg
+
+  const adminId = req.user?.id
+  if (!adminId) throw new Error('No User')
+
+  const exercise = await prisma.exercise.findUnique({
+    where: {
+      id
+    }
+  })
+
+  if (!get(exercise, 'flaggedAt')) {
+    throw new Error('Exercise is already not flagged')
+  }
+
+  if (!isAdmin(req)) {
```

Comment:
move this check to before the `findUnique` because if req is not admin, we can save a `findUnique` query.